test/buf-ring-nommap.c

```diff
@@ -41,6 +41,8 @@ int main(int argc, char *argv[])
 	if (posix_memalign(&ring_mem, 16384, 16384))
 		return T_EXIT_FAIL;
 
+	memset(ring_mem, 0, 16384);
+
 	p.flags = IORING_SETUP_NO_MMAP;
 	ret = io_uring_queue_init_mem(1, &ring, &p, ring_mem, 16384);
```

Comment:
Thanks for debugging this...
Was going to ask why `io_uring_queue_init_mem()` shouldn't handle it, but the pre-zeroed requirement is already carried in the manpage: `The buffer passed to io_uring_queue_init_mem(3) must already be zeroed.`.